scaleway/resource_instance_server.go

```diff
@@ -678,6 +679,10 @@ func resourceScalewayInstanceServerUpdate(ctx context.Context, d *schema.Resourc
 			if err != nil {
 				return diag.FromErr(err)
 			}
+			//we wait to ensure to not detach the new ip.
+			if newIPID != "" {
```

Comment:
@Monitob could you amend this PR to have a wait based workflow? We should be able to see whether the IP got a status or if the server got a status updated with the IP. Checking with the instance team could be helpful.

Reply:
This resource has been updated to wait a stable state.